R/plotstyle.R

```diff
@@ -1,174 +1,194 @@
+#'   produced (to produce plots with all available entities leave the \code{...}
+#'   entry empty!)
+#' @param verbosity Set to 1 if you want to know for which unknown entities
+#'   plotstyle brewed colors
+#' @param regexp If \code{TRUE}, match entities as regular expressions. Matching
+#'   entities are expanded, non-matching entities are returned as the original
+#'   expression.  Does not generate default color maps. Implies \code{plot =
+#'   FALSE} and \code{verbosity = 0}.
 #' @return Plot styles for given entities
+#' @section Colors for unknown entities:
+#' \if{html}{\figure{colors.png}{options: width="100\%"}}
 #' @author David Klein, Jan Philipp Dietrich
 #' @seealso \code{\link{plotstyle.add}}
 #' @examples
-#' entities <- c("AFR","AAA","AFR","UNKNOWN_ELEMENT2")
+#' entities <- c("AFR", "AAA", "AFR", "UNKNOWN_ELEMENT2")
 #' plotstyle(entities)
-#' unknown=data.frame(row.names=c("AAA","UNKNOWN_ELEMENT2"),
-#'                    color=c("#123456","#345678"),
-#'                    legend=c("l_AAA","l_UNKNOWN_ELEMENT2"))
-#' plotstyle(entities,unknown=unknown)
-#' plotstyle(entities,out="legend")
-#' plotstyle(entities,out="all")
-#' 
+#' unknown <- data.frame(row.names = c("AAA", "UNKNOWN_ELEMENT2"),
+#'                    color = c("#123456", "#345678"),
+#'                    legend = c("l_AAA", "l_UNKNOWN_ELEMENT2"))
+#' plotstyle(entities, unknown = unknown)
+#' plotstyle(entities, out = "legend")
+#' plotstyle(entities, out = "all")
+#'
 #' # search for all 'Final Energy Biomass' entities
-#' plotstyle('^Final Energy\\|.*Biomass', regexp = TRUE)
-#' 
+#' plotstyle("^Final Energy\\|.*Biomass", regexp = TRUE)
+#'
 #' # search for all three-letter entities (a.k.a. regions)
-#' plotstyle('^[A-Z]{3}$', regexp = TRUE, out = 'all')
+#' plotstyle("^[A-Z]{3}$", regexp = TRUE, out = "all")
 #' @export
 #' @importFrom utils read.csv2
 #' @importFrom grDevices rgb
 #' @importFrom RColorBrewer brewer.pal
 #' @importFrom ggplot2 ggplot geom_bar coord_flip theme element_blank labs aes
 #' @importFrom grDevices colorRampPalette
 
-plotstyle <- function(..., out="color", unknown=NULL, plot=FALSE, verbosity = getOption("plotstyle.verbosity"), 
+plotstyle <- function(..., out = "color", unknown = NULL, plot = FALSE, verbosity = getOption("plotstyle.verbosity"),
                       regexp = FALSE) {
-  
-  luplot<-list()
-  luplot$plotstyle <- read.csv2(system.file("extdata","plotstyle.csv",package = "mip"),stringsAsFactors = F,row.names=1)
-  
-  if(is.null(out)) {
+
+  luplot <- list()
+  luplot$plotstyle <- read.csv2(
+    system.file("extdata", "plotstyle.csv", package = "mip"),
+    stringsAsFactors = FALSE,
+    row.names = 1)
+
+  if (is.null(out)) {
     out <- "color"
-  } else if(!(out %in% c(names(luplot$plotstyle),"all"))) {
-    stop('Unknown style type "',out,'"!')
+  } else if (!(out %in% c(names(luplot$plotstyle), "all"))) {
+    stop('Unknown style type "', out, '"!')
   }
-  
```

Comment:
Is there a reason you don't allow the gray levels anymore?

Reply:
The function `.makegray()` is never used in the code. So I removed it.